pandas/io/json/_normalize.py

```diff
@@ -90,20 +157,29 @@ def nested_to_record(
             else:
                 newkey = prefix + sep + k
 
-            # flatten if type is dict and
-            # current dict level  < maximum level provided and
-            # only dicts gets recurse-flattened
-            # only at level>1 do we rename the rest of the keys
-            if not isinstance(v, dict) or (
-                max_level is not None and level >= max_level
+            # flatten if
+            # current dict level <= maximum level provided and
+            # current keypath matches the config in use_keys
+            # only dicts gets recurse-flatten
+            if (
+                is_key_match(newkey, use_keys)
```

Comment:
I don't think this validation is necessary for the change; can you remove `is_key_match`?